src/net_processing.cpp

```diff
@@ -468,6 +468,44 @@ static CNodeState *State(NodeId pnode) EXCLUSIVE_LOCKS_REQUIRED(cs_main) {
     return &it->second;
 }
 
+/**
+ * Data stucture for an individual peer. This struct is not protected by
+ * cs_main since it does not contain validation-critical data.
+ *
+ * Memory is owned by shared pointers and this object is destructed when
+ * the refcount drops to zero (in FinalizeNode()).
+ *
+ * TODO: move most members from CNodeState to this structure.
+ * TODO: move remaining application-layer data members from CNode to this structure.
+ */
+struct Peer {
+    /** Same id as the CNode object for this peer */
+    NodeId m_id;
```

Comment:
```suggestion
    const NodeId m_id;
```

in commit d60a16bb3f58fcb9422c277256b3c3a31cab3b2f

Any reason to not make this const? Having this mutable would violate the assumption that the id is unique per peer.

Reply:
You're right. This should be const. Fixed